File: Source/Back-end/src/main/java/org/elis/manutenzione/dto/request/FiltroManutenzione.java
package org.elis.manutenzione.dto.request;

import lombok.Data;

/**
 * DTO che rappresenta i filtri per la ricerca delle manutenzioni.
 */
@Data
public class FiltroManutenzione {

    /**
     * Descrizione della manutenzione.
     */
    private String descrizione;

    /**
     * Data di richiesta della manutenzione.
     */
    private String dataRichiesta;

    /**
     * Data di riparazione della manutenzione.
     */
    private String dataRiparazione;

    /**
     * Data prevista per la riparazione della manutenzione.
     */
    private String dataPrevista;

    /**
     * Priorità della manutenzione.
     */
    private int priorita;

    /**
     * Email del manutentore assegnato alla manutenzione.
     */
    private String manutentore;

    /**
     * Stato della manutenzione (es. In attesa, In corso, Completata).
     */
    private String statoManutenzione;

    /**
     * Email dell'utente che ha richiesto la manutenzione.
     */
    private String utenteRichiedente;

    /**
     * Nome della manutenzione.
     */
    private String nome;
}

File: Source/Front-end/front-end/src/app/model/request/filtro-manutenzione.ts
export class FiltroManutenzione {
  descrizione: string;
  dataRichiesta: string;
  dataRiparazione: string;
  dataPrevista: string;
  priorita: number;
  manutentore: string;
  statoManutenzione: string;
  utenteRichiedente: string;
  nome: string;

  constructor(descrizione: string, dataRichiesta: string, dataRiparazione: string, dataPrevista: string, priorita: number, manutentore: string, statoManutenzione: string, utenteRichiedente: string, nome: string)
  constructor();


  constructor(descrizione?: string, dataRichiesta?: string, dataRiparazione?: string, dataPrevista?: string, priorita?: number, manutentore?: string, statoManutenzione?: string, utenteRichiedente?: string, nome?: string) {
    this.descrizione = descrizione ?? "";
    this.dataRichiesta = dataRichiesta ?? "";
    this.dataRiparazione = dataRiparazione ?? "";
    this.dataPrevista = dataPrevista ?? "";
    this.priorita = priorita ?? 0;
    this.manutentore = manutentore ?? "";
    this.statoManutenzione = statoManutenzione ?? "";
    this.utenteRichiedente = utenteRichiedente ?? "";
    this.nome = nome ?? "";
  }

}
